dashboard/app/models/levels/level.rb

```diff
@@ -1013,8 +1013,23 @@ def skill_identifiers
     skills.map {|skill| {id: skill.id, key: skill.key}}
   end
 
-  def skill_keys
-    skills.pluck(:key)
+  def remove_skill_key(skill_key)
+    return unless Rails.application.config.levelbuilder_mode
```

Comment:
if you are going to check for this case, I would recommend raising instead, here and below. it likely means a programming error has been made and should not be silently ignored.